Publication/Publication/Publication.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+public enum PublicationType { Misc, Book, Magazine, Article };
+
+public abstract class Publication
+{
+    private bool published = false;
+    private DateTime datePublished;
+    private int totalPages;
+
+    public string Publisher { get; }
+    public string Title { get; }
+    public PublicationType Type { get; }
+    public string CopyrightName { get; private set; }
+    public int CopyrightDate { get; private set; }  // int as only year value
+    public int Pages
+    {
+        get { return totalPages; }
+        set
+        {
+            if (value <= 0)
+                throw new ArgumentOutOfRangeException("The number of pages must be greater than zero.");
+            totalPages = value;
+        }
+    }
+
+    /// <summary>
+    /// Constructor for Publication; Checks for non-empty title and publisher
+    /// </summary>
+    /// <param name="title">Publication's Title</param>
+    /// <param name="publisher">Publication's Publisher</param>
+    /// <param name="type">Type of publication eg. book, magazine etc</param>
+    public Publication(string title, string publisher, PublicationType type)
+    {
+        if (String.IsNullOrWhiteSpace(publisher))
+            throw new ArgumentException("The publisher is required.");
+        Publisher = publisher;
+
+        if (String.IsNullOrWhiteSpace(title))
+            throw new ArgumentException("The title is required.");
+        Title = title;
+
+        Type = type;
+    }
+
+    /// <summary>
+    /// Getter for publication date - returns "NYP" if not published
+    /// </summary>
+    /// <returns>Date published (as string - short date format) or "NYP" if not published</returns>
```

Comment:
In the <returns> you can use the <see cref=""> Syntax. See https://docs.microsoft.com/en-us/dotnet/csharp/codedoc for more information